File: b0204trlight/app/src/main/java/com/example/b0204trlight/MainActivity.java
package com.example.b0204trlight;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;

public class MainActivity extends AppCompatActivity {

    LinearLayout b1, b2, b3;
    Button button1;
    private boolean start_stop = false;
    private int counter = 0;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        b1 = findViewById(R.id.bulb1);
        b2 = findViewById(R.id.bulb2);
        b3 = findViewById(R.id.bulb3);
        button1 = findViewById(R.id.button1);

        b1.setBackgroundColor(Color.GREEN);

    }


    public void onClickStart(View view) {
        if(!start_stop){
            start_stop = true;
            button1.setText(getResources().getText(R.string.stop));

            new Thread(new Runnable() {
                @Override
                public void run() {
//                    b1.setBackgroundColor(getResources().getColor(R.color.teal_200));
                    while (start_stop){
                        counter++;
                        switch (counter){
                            case 1:
                                b1.setBackgroundColor(Color.GREEN);
                                b2.setBackgroundColor(getResources().getColor(R.color.teal_200));
                                b3.setBackgroundColor(getResources().getColor(R.color.teal_200));
                                break;
                            case 2:
                                b1.setBackgroundColor(getResources().getColor(R.color.teal_200));
                                b2.setBackgroundColor(Color.YELLOW);
                                b3.setBackgroundColor(getResources().getColor(R.color.teal_200));
                                break;
                            case 3:
                                b1.setBackgroundColor(getResources().getColor(R.color.teal_200));
                                b2.setBackgroundColor(getResources().getColor(R.color.teal_200));
                                b3.setBackgroundColor(Color.RED);
                                counter = 0;
                                break;
                        }
                        try{
                            Thread.sleep(1000);
                        }catch (InterruptedException e){
                            e.printStackTrace();
                        }
                    }
                }
            }).start();


        }else {
            start_stop = false;
            button1.setText(getResources().getText(R.string.start));
        }

    }
}

File: l2012var/app/src/main/java/com/example/l2012var/MainActivity.kt
package com.example.l2012var

import android.graphics.Color
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Layout
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout

class MainActivity : AppCompatActivity() {

    private var textView1: TextView? = null
    private var layout1: ConstraintLayout? = null
    private var start: Boolean = false
    private var counter: Int = 0

    private var n1:Byte = 1
    private var n2:Short = 2
    private var n3:Int = 3
    private var n4:Long = 4
    private var n5:Float = 5.0f
    private var n6:Double = 6.0
    private var b7:Boolean = true
    private var s8:Char = '8'
    private var s9:String = "99 9 99"


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        textView1 = findViewById(R.id.textView1)
        layout1 = findViewById(R.id.layout1)

        Thread{
            start = true
            while (start){
                Thread.sleep(500)

                runOnUiThread(){
                    textView1?.setText(counter.toString())
                    if(counter == 5){
                        layout1?.setBackgroundColor(Color.GRAY)
                    }else if(counter == 10){
                        layout1?.setBackgroundColor(Color.YELLOW)
                    }
                    counter++
                }

            }
        }.start()
    }

    override fun onDestroy() {
        super.onDestroy()
        start = false
    }
}